format/src/format.rs

```diff
@@ -863,17 +863,54 @@ impl FormatString {
     }
 
     fn parse_part_in_brackets(text: &str) -> Result<FormatPart, FormatParseError> {
-        let parts: Vec<&str> = text.splitn(2, ':').collect();
+        let mut chars = text.chars().peekable();
+
+        let mut left = String::new();
+        let mut right = String::new();
+
+        let mut split = false;
+        let mut inside_brackets = false;
+
+        while let Some(char) = chars.next() {
+            if char == '[' {
+                inside_brackets = true;
+
+                if split {
+                    right.push(char);
+                } else {
+                    left.push(char);
+                }
+
+                while let Some(next_char) = chars.next() {
+                    if split {
+                        right.push(next_char);
+                    } else {
+                        left.push(next_char);
+                    }
+
+                    if next_char == ']' {
+                        inside_brackets = false;
+                        break;
+                    }
+                    if chars.peek().is_none() {
+                        return Err(FormatParseError::MissingRightBracket);
+                    }
+                }
+            } else if char == ':' && !split && !inside_brackets {
+                split = true;
+            } else if split {
+                right.push(char);
+            } else {
+                left.push(char);
+            }
+        }
```

Comment:
Adding `selected` makes it easier to read the code 

```suggestion
        let mut split = false;
        let mut selected = &mut left;
        let mut inside_brackets = false;

        while let Some(char) = chars.next() {
            if char == '[' {
                inside_brackets = true;

                selected.push(char);

                while let Some(next_char) = chars.next() {
                    selected.push(next_char);

                    if next_char == ']' {
                        inside_brackets = false;
                        break;
                    }
                    if chars.peek().is_none() {
                        return Err(FormatParseError::MissingRightBracket);
                    }
                }
            } else if char == ':' && !split && !inside_brackets {
                split = true;
                selected = &mut right;
            } else {
                selected.push(char);
            }
        }
```